src/plugins/controls/server/options_list/options_list_cluster_settings_route.ts

```diff
@@ -39,6 +40,17 @@ export const setupOptionsListClusterSettingsRoute = ({ http }: CoreSetup) => {
           },
         });
       } catch (e) {
+        if (e instanceof errors.ResponseError && e.body.error.type === 'security_exception') {
```

Comment:
Should we also show a toast, notifying users that they are missing permissions? We do something like this in maps https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/classes/sources/es_search_source/util/load_index_settings.ts#L46 when fetching index.max_result_window, where we use a default but notify users when there is a permission problem.

Reply:
I'm honestly not sure about this. Yes there is a permissions problem, which should  _probably_ bubble up - but we don't really want to tie the usage of Controls to the `monitor` privilege so closely. If there was a toast, it would basically tell the end-user - not the author in this case because they usually have the right permissions - that in order to use controls they need the monitor privilege on the index they use.

If there was a way to warn only the author when they were setting up the roles that would be okay with me, but I don't think we have that kind of mechanism.

Reply:
Maybe a warning on the server-side? Is that a pattern that we use?

Reply:
I think I'm going to merge this as is, but if it does cause that problem we're aware of - where `allow_expensive_queries` is off and the user doesn't have permissions to check - we can revisit this conversation and show a toast, or do some other less intrusive warning.